src/ae.c

```diff
@@ -333,7 +333,7 @@ static int processTimeEvents(aeEventLoop *eventLoop) {
             processed++;
             now = getMonotonicUs();
             if (retval != AE_NOMORE) {
-                te->when = now + retval * 1000;
+                te->when = now + (uint64_t)retval * 1000;
```

Comment:
seems more appropriate to cast it to `long long`, or `monotime` (that's the type of `now`)
```suggestion
                te->when = now + (monotime)retval * 1000;
```